CMakeLists.txt

```
@@ -940,44 +940,18 @@ if(NINTENDO_3DS)
 endif()
 
 if(CPACK)
-  if(NOT SYSTEM_BITS MATCHES x64 AND WIN32)
-    set(CPACK_PACKAGE_FILE_NAME "devilutionx")
-    set(CPACK_PACKAGE_NAME ${project_name})
-    set(CPACK_GENERATOR "ZIP")
-    set(CPACK_STRIP_FILES TRUE)
-    install(TARGETS ${BIN_TARGET} DESTINATION .)
-    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/windows/README.txt"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/CharisSILB.ttf"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/devilutionx.mpq"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2_ttf-2.0.15/i686-w64-mingw32/bin/libfreetype-6.dll"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2-2.0.14/i686-w64-mingw32/bin/SDL2.dll"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2_ttf-2.0.15/i686-w64-mingw32/bin/SDL2_ttf.dll"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2_ttf-2.0.15/i686-w64-mingw32/bin/zlib1.dll"
-      DESTINATION "."
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/LICENSE.CharisSILB.txt"
-      DESTINATION "LICENSE"
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2_ttf-2.0.15/i686-w64-mingw32/bin/LICENSE.freetype.txt"
-      DESTINATION "LICENSE"
-    )
-    install(FILES "${PROJECT_SOURCE_DIR}/SDL2-2.0.14/README-SDL.txt"
-      DESTINATION "LICENSE"
-    )
+  if(WIN32)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+      set(SDL2_WIN32_DLLS_DIR "${CMAKE_BINARY_DIR}")
+      set(SDL2_WIN32_LICENSES_DIR "${PROJECT_SOURCE_DIR}/Packaging/resources")
+    else()
+      set(SDL2_WIN32_DLLS_DIR "${SDL2_EXEC_PREFIX}/bin")
+      set(SDL2_WIN32_LICENSES_DIR "${SDL2_EXEC_PREFIX}/bin")
+    endif()
+
+    file(GLOB _SDL2_ALL_DLLS "${SDL2_EXEC_PREFIX}/bin/*.dll")
+    file(GLOB _SDL2_ALL_LICENSES "${SDL2_EXEC_PREFIX}/bin/LICENSE*.txt")

```

I don't think this works for MinGW since the dlls and license files don't all come from the same directory. Anyway, here's a suggestion with variable names to match the ones you use above and below.

```suggestion
    file(GLOB SDL2_WIN32_ALL_DLLS "${SDL2_WIN32_DLLS_DIR}/*.dll")
    file(GLOB SDL2_WIN32_ALL_LICENSES "${SDL2_WIN32_LICENSES_DIR}/LICENSE*.txt")
```

---

It happens that on WSL (see my comment above) they do: that directory seems to hold a "redistributable" version of SDL dll's along with licensing information. I used that directory to have the current licenses (rather than possibly outdated ones for older SDL2 builds). The ones I added to "Packaging/resources" are a specific fix for MSVC builds, since locating the licenses (at least for vcpkg) seemed to be quite a challenge.

In short---in my opinion keeping the same SDL2_EXEC_PREFIX would make a little more sense on Linux builds.

---

Sorry, but your insistence that it does work does not change the reality that it didn't work on CI. Furthermore, if you don't fix the variables based on my suggestion, then it won't work properly for VS either.

---

Wow, I seem to have screwed up again, sorry. This time it's the naming. I'll update the PR shortly.

---

Just a little explanation: I have two copies of devilutionX. Copy-pasting and a last-minute deciision to make names more consistent with other CMake variables was a mistake. Windows artifacts now look okay ^(. .)^

---

> Sorry, but your insistence that it does work does not change the reality that it didn't work on CI

What does CI stand for?

---

Continuous Integration

---

> Just a little explanation: I have two copies of devilutionX. Copy-pasting and a last-minute deciision to make names more consistent with other CMake variables was a mistake. Windows artifacts now look okay ^(. .)^

Happenes a lot to me as well, that's what the CI is for 🙂

---

I just clicked the "Resolve conversation" button in the branch above, and now it is gone. Have I done something horrible?

---

> I just clicked the "Resolve conversation" button in the branch above, and now it is gone. Have I done something horrible?

No, it's just been collapsed. You can expand it again to read the resolved conversation by clicking `Show resolved` on the far right.